Quantum-Control-Applications/Superconducting/Single Flux Tunable Transmon/SSB mixer for readout/macros.py

```diff
@@ -10,34 +10,55 @@
 # QUA macros #
 ##############
 
-# Macro for performing a single shot active reset.
-def active_reset_single(threshold=None, Ig=None):
-    """Macro performing a single shot active reset.
 
-    :param threshold: threshold for the 'I' quadrature discriminating between ground and excited state.
-    :param Ig: A QUA variable for the information in the `I` quadrature. Should be of type `Fixed`. If not given, a new
-        variable will be created
-    :return: A QUA variable for the information in the `I` quadrature.
+def reset_qubit(method, **kwargs):
     """
-    if Ig is None:
-        Ig = declare(fixed)
-    if threshold is None:
-        raise Exception("A threshold needs to be specified to perform active reset.")
+    Macro to reset the qubit state.
 
-    measure(
-        "readout",
-        "resonator",
-        None,
-        dual_demod.full("rotated_cos", "out1", "rotated_sin", "out2", Ig),
-    )
-    # Perform active feedback
-    # Use the single conditional play statement for integrating active reset in other protocols
-    play("pi", "qubit", condition=(Ig < threshold))
-    return Ig
+    If method is 'cooldown', then the variable cooldown_time (in clock cycles) must be provided as a python integer > 4.
+
+    **Example**: reset_qubit('cooldown', cooldown_times=500)
 
+    If method is 'active', then 3 parameters are available as listed below.
 
-# Macro for performing active reset until succesfull for a gicen number of tries.
-def active_reset_until_success(threshold, max_tries=1, Ig=None):
+    **Example**: reset_qubit('active', threshold=-0.003, max_tries=1)
```

Comment:
I would give the example either without max_tries or with max_tries>1 (3?)

Reply:
Ok